### PR TITLE
Adds department wardrobe vendors to the code and to Boxstation.

### PR DESCRIPTION
**Pull request by Firecage**
This PR adds department wardrobe vendors to the code, so that if they get added to the maps, people will be able to dress up more easily without having the clutter of one big pile of stuff in a locker.

:cl: Firecage
Add: Replaces wardrobe lockers on Box with Wardrobe Vending Machine.
/:cl:

atmosdrobe_0
atmosdrobe-broken_0
atmosdrobe-off_0
atmosdrobe-panel_0
bardrobe_0
bardrobe-broken_0
bardrobe-off_0
bardrobe-panel_0
cargodrobe_0
cargodrobe-broken_0
cargodrobe-off_0
cargodrobe-panel_0
chapdrobe_0
chapdrobe-broken_0
chapdrobe-off_0
chapdrobe-panel_0
chefdrobe_0
chefdrobe-broken_0
chefdrobe-off_0
chefdrobe-panel_0
chemdrobe_0
chemdrobe-broken_0
chemdrobe-off_0
chemdrobe-panel_0
curadrobe_0
curadrobe-broken_0
curadrobe-off_0
curadrobe-panel_0
engidrobe_0
engidrobe-broken_0
engidrobe-off_0
engidrobe-panel_0
genedrobe_0
genedrobe-broken_0
genedrobe-off_0
genedrobe-panel_0
hydrobe_0
hydrobe-broken_0
hydrobe-off_0
hydrobe-panel_0
janidrobe_0
janidrobe-broken_0
janidrobe-off_0
janidrobe-panel_0
lawdrobe_0
lawdrobe-broken_0
lawdrobe-off_0
lawdrobe-panel_0
medidrobe_0
medidrobe-broken_0
medidrobe-off_0
medidrobe-panel_0
robodrobe_0
robodrobe-broken_0
robodrobe-off_0
robodrobe-panel_0
scidrobe_0
scidrobe-broken_0
scidrobe-off_0
scidrobe-panel_0
secdrobe_0
secdrobe-broken_0
secdrobe-off_0
secdrobe-panel_0
virodrobe_0
virodrobe-broken_0
virodrobe-off_0
virodrobe-panel_0

### CODE REVIEW

**Comment by 81Denton** [comment]
Very nice! I'm digging the cheeky names and product ads.

File: code/modules/vending/wardrobes.dm [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+
+/obj/item/vending_refill/wardrobe/engi_wardrobe
+	machine_name = "EngiDrobe"
+	charges = list(7, 0, 0)
+	init_charges = list(7, 0, 0)
+
+/obj/machinery/vending/wardrobe/atmos_wardrobe
+	name = "AtmosDrobe"
+	desc = "This relatively unknown vending machine delivers clothing for Atmospherics Technicions, an equally unknown job."
+	product_ads = "Get your inflammible clothing right here!!!"
```
Comment (81Denton) [inline code/modules/vending/wardrobes.dm]: inflammable

**Comment by 81Denton** [comment]
>I originally intended to add these to boxstation, but the map merger won't work

I'll add them for you after the PR gets merged

File: code/modules/vending/wardrobes.dm (hunk 1 above)
Comment (Firecage) [reply code/modules/vending/wardrobes.dm]: Hope you get the joke with inflammable though.

**Comment by ExcessiveUseOfCobblestone** [comment]
Could you make it so you can add items to the machine if they're part of the list? IE If I want to trade out for medical scrubs, I can put my jumpsuit in the machine as well.

Also the access thing is kinda meh, as they exist for people to subterfuge with. If you insist on doing it though, please use[ the defines.](https://github.com/tgstation/tgstation/blob/11d96d9550d05e784f84231953a2abd7c1c9166c/code/__DEFINES/access.dm)

**Comment by Firecage** [comment]
Mmm, naturally if people would prefer it without the access addition, I could remove that. Though if people are in support of it, then sure, will use the better defines for them. Do wonder if req_access_txt would accept the worded defines though, since I merely used the numbers as all the other access locked vending machines does.

**Comment by ExcessiveUseOfCobblestone** [comment]
I think it causes unneeded balance discussion for an otherwise good PR.

the link I put are the numerical value defines so they should be fine. If you want to append the defines to all vending machines though then be my guest :^)

**Comment by Firecage** [comment]
Eh, might be a better idea to remove them, since the specific wardrobe lockers these replaces didn't have access in the first place. Now naturally if people wants these to go further(cough, vendor for the 9 sec lockers with gear, cough), then those should probably have access req.

**Comment by Firecage** [comment]
The big thing will be to get someone to make unique sprites for them though, since as the PR says, they use the default autodrobe sprites for now. Sure, its a fitting sprite, but it would confuse people.

File: code/modules/vending/wardrobes.dm (hunk 1 above)
Comment (ike709) [reply code/modules/vending/wardrobes.dm]: The English language really pisses me off sometimes; this is one of those times.

File: code/modules/vending/wardrobes.dm [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+	refill_canister = /obj/item/vending_refill/wardrobe/sec_wardrobe
+
+/obj/item/vending_refill/wardrobe/sec_wardrobe
+	machine_name = "SecDrobe"
+	charges = list(8, 0, 0)
+	init_charges = list(8, 0, 0)
+
+/obj/machinery/vending/wardrobe/medi_wardrobe
+	name = "\improper MediDrobe"
+	desc = "A vending machine rumoured to be capable in dispensing clothing for medical personnel."
```
Comment (Mickyan) [inline code/modules/vending/wardrobes.dm]: "capable of" would be more appropriate

File: code/modules/vending/wardrobes.dm [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+	refill_canister = /obj/item/vending_refill/wardrobe/medi_wardrobe
+
+/obj/item/vending_refill/wardrobe/medi_wardrobe
+	machine_name = "MediDrobe"
+	charges = list(10, 0, 0)
+	init_charges = list(10, 0, 0)
+
+/obj/machinery/vending/wardrobe/engi_wardrobe
+	name = "EngiDrobe"
+	desc = "A vending machine reknowned for vending industrial grade clothing."
```
Comment (Mickyan) [inline code/modules/vending/wardrobes.dm]: *renowned

File: code/modules/vending/wardrobes.dm [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+	refill_canister = /obj/item/vending_refill/wardrobe/engi_wardrobe
+
+/obj/item/vending_refill/wardrobe/engi_wardrobe
+	machine_name = "EngiDrobe"
+	charges = list(7, 0, 0)
+	init_charges = list(7, 0, 0)
+
+/obj/machinery/vending/wardrobe/atmos_wardrobe
+	name = "AtmosDrobe"
+	desc = "This relatively unknown vending machine delivers clothing for Atmospherics Technicions, an equally unknown job."
```
Comment (Mickyan) [inline code/modules/vending/wardrobes.dm]: *Technicians

Comment (Firecage) [reply code/modules/vending/wardrobes.dm]: And this, my friends, is why I am not to be trusted with descriptions.

**Comment by Arianya** [comment]
@ExcessiveUseOfCobblestone So long as these use IDs to check for access it should be fine

Bear in mind thing like sec jumpsuits have ingrained balance benefits (armour) which make it desirable to lock them behind the required access. If someone has access to the sec lockers, they have access to the sec autodrobe

**Comment by Firecage** [comment]
Well, I mean...I could.

**Comment by Firecage** [comment]
Cobby, I wonder, should those formal sec uniforms, the ones locked in a crate outside armory, also be added to the sec vendor?

**Comment by ExcessiveUseOfCobblestone** [comment]
Not sure why they're locked but if there's no outstanding issue that would be neat (if there is you can prob just toss it under premium items). We also have grey variants i'm sure Nilons would love me forever if I asked you to include those.

**Comment by Firecage** [comment]
I do kind of wonder why they are locked in a crate...which is behind a bunch of portable flashes, almost like someone didn't WANT people to use them.

**Comment by Alterist** [comment]
Woah wait, this is ready to merge? I came here cause I was the one who suggested one the the RoboDrobe ads and wanted to make another suggestion. (I'll get to that later) But even though I'm not a coder I could see an issue with the code right now.
```
+/obj/item/vending_refill/wardrobe/bar_wardrobe
+	machine_name = "CuraDrobe"
+	charges = list(8, 0, 0)
+	init_charges = list(8, 0, 0)
```
```
+/obj/item/vending_refill/wardrobe/hydro_wardrobe
+	machine_name = "SciDrobe"
+	charges = list(6, 0, 0)
+	init_charges = list(6, 0, 0)
```
I'm pretty sure that is not what you intended, I'm not quite sure what this error would even do, but I'm pretty sure it's wrong. There are also a few more errors:
```
+/obj/machinery/vending/wardrobe/law_wardrobe
+	name = "LawDrobe"
+	desc = "Objection! This wardrobe dispenses the rule of law... and lawyer clothing."
+	product_ads = "OBJECTION! Get the rule of law for yourself!"
+	vend_reply = "Thank you for using the JaniDrobe!"
```
I suppose this is correct in a way, Lawyers are as useful as Janitors in an usual round.
`product_ads = "Are you being botheres by cultists or pesky reverants? Then come and dress like the holy man!"`
it's "bothered" and "revenants"
`+	product_ads = " Virusses getting you down? Then upgrade to sterilized clothing today!"`
it's "viruses" Really, a simple spellcheck would do wonders. sorry for this massive post, It's my first time commenting on a pull, I'll put the ad suggestions on another pot then

**Comment by Firecage** [comment]
Dag nabbit, and that is why I should NOT write code after drinking a beer. Good catch.

File: code/modules/vending/wardrobes.dm [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+	icon_state = "clothes"
+	icon_deny = "clothes-deny"
+
+/obj/item/vending_refill/wardrobe
+	icon_state = "refill_clothes"
+
+/obj/machinery/vending/wardrobe/sec_wardrobe
+	name = "\improper SecDrobe"
+	desc = "A vending machine for security and security-related clothing!"
+	product_ads = "Beat perps in style!; It's red so you can't see the blood!; You have the right to be fashionable!; Now you can be the fashion police you always wanted to be!"
```
Comment (ShizCalev) [inline code/modules/vending/wardrobes.dm]: remove the spaces after the `;`'s

File: code/modules/vending/wardrobes.dm [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+
+/obj/item/vending_refill/wardrobe/medi_wardrobe
+	machine_name = "MediDrobe"
+	charges = list(10, 0, 0)
+	init_charges = list(10, 0, 0)
+
+/obj/machinery/vending/wardrobe/engi_wardrobe
+	name = "EngiDrobe"
+	desc = "A vending machine renowned for vending industrial grade clothing."
+	product_ads = "Guaranteed to protect your feet from industrial accidents!; Afraid of radiation? Then wear yellow!"
```
Comment (ShizCalev) [inline code/modules/vending/wardrobes.dm]: same

File: code/modules/vending/wardrobes.dm [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+
+/obj/item/vending_refill/wardrobe/cargo_wardrobe
+	machine_name = "CargoDrobe"
+	charges = list(5, 0, 0)
+	init_charges = list(5, 0, 0)
+
+/obj/machinery/vending/wardrobe/robo_wardrobe
+	name = "RoboDrobe"
+	desc = "A vending machine designed to dispense clothing known only to roboticists."
+	product_ads = "You turn me TRUE, use defines!, 0100001101101100011011110111010001101000011010010110111001100111001000000110100001100101011100100110010100100001"
```
Comment (ShizCalev) [inline code/modules/vending/wardrobes.dm]: `!,`?

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
Do we even use ads?

**Comment by Firecage** [comment]
Yah, I have heard the vending machines spout them now and then.

**Comment by DaxDupont** [comment]
I don't think these should replace closets though, I'd like to grab clothes to disguise when there's no power or drag closets to maint to dress there.

**Comment by Firecage** [comment]
Dax, well, you can make vending machines drop their contents without power.

**Comment by Alterist** [comment]
I'm looking at the code for other vending machines and I need to ask: what's the difference between product_slogans and product_ads?
also:
`+	product_ads = "Do you love soil? Then buy our clothes!,Get outfits to match your green thumb here!"`
you placed a comma instead of a semicolon

**Comment by JStheguy** [comment]
dissapoint
:angry:

**Comment by Firecage** [comment]
JS, I am failing to see why you are making an angry face?

File: code/modules/cargo/packs.dm [hunk 8]
```diff
@@ -1714,6 +1714,29 @@
...
+					/obj/item/vending_refill/wardrobe/hydro_wardrobe,
+					/obj/item/vending_refill/wardrobe/curator_wardrobe,
+					/obj/item/vending_refill/wardrobe/bar_wardrobe,
+					/obj/item/vending_refill/wardrobe/chef_wardrobe,
+					/obj/item/vending_refill/wardrobe/jani_wardrobe,
+					/obj/item/vending_refill/wardrobe/law_wardrobe,
+					/obj/item/vending_refill/wardrobe/chap_wardrobe,
+					/obj/item/vending_refill/wardrobe/chem_wardrobe,
+					/obj/item/vending_refill/wardrobe/gene_wardrobe,
+					/obj/item/vending_refill/wardrobe/viro_wardrobe)
```
Comment (ninjanomnom) [inline code/modules/cargo/packs.dm]: Please split this up into departments

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/game/objects/items/circuitboards/machine_circuitboards.dm [hunk 9]
```diff
@@ -225,6 +225,23 @@
...
+							/obj/machinery/vending/wardrobe/hydro_wardrobe = "HyDrobe",
+							/obj/machinery/vending/wardrobe/curator_wardrobe = "CuraDrobe",
+							/obj/machinery/vending/wardrobe/bar_wardrobe = "BarDrobe",
+							/obj/machinery/vending/wardrobe/chef_wardrobe = "ChefDrobe",
+							/obj/machinery/vending/wardrobe/jani_wardrobe = "JaniDrobe",
+							/obj/machinery/vending/wardrobe/law_wardrobe = "LawDrobe",
+							/obj/machinery/vending/wardrobe/chap_wardrobe = "ChapDrobe",
+							/obj/machinery/vending/wardrobe/chem_wardrobe = "ChemDrobe",
+							/obj/machinery/vending/wardrobe/gene_wardrobe = "GeneDrobe",
+							/obj/machinery/vending/wardrobe/viro_wardrobe = "ViroDrobe",
```
Comment (ninjanomnom) [inline code/game/objects/items/circuitboards/machine_circuitboards.dm]: It's unnecessary in this pr but this list really should be generated

Comment (Firecage) [reply code/game/objects/items/circuitboards/machine_circuitboards.dm]: What exactly do you mean by generated?

Comment (ninjanomnom) [reply code/game/objects/items/circuitboards/machine_circuitboards.dm]: The list should not have to be manually updated and just generate the list from the type path

Comment (Firecage) [reply code/game/objects/items/circuitboards/machine_circuitboards.dm]: Mmm, it alright if I leave that for another PR?

Comment (ninjanomnom) [reply code/game/objects/items/circuitboards/machine_circuitboards.dm]: >It's unnecessary in this pr but this list really should be generated

**Review by ninjanomnom — APPROVED** [review]

**Comment by 81Denton** [comment]
@Firecage I‘ll open a PR to map them in later today or tomorrow. Do you have any preferences for their locations?

**Comment by Firecage** [comment]
@81Denton Note, I already added them to box. For the other maps, perhaps just replace the wardrobe lockers with them for now, so people know where they are located?

**Pull request merged**